lib/Lib.Buffer.fsti

```diff
@@ -42,6 +42,10 @@ unfold let ibuffer (a:Type0) = buffer_t IMMUT a
 (** Const buffer. Either one of the two above. Extracts as `const a*`. *)
 unfold let cbuffer (a:Type0) = buffer_t CONST a
 
+(** Global buffers are const immutable *)
+unfold let gbuffer (a:Type0) = c:cbuffer a{CB.qual_of c == CB.IMMUTABLE /\
+					   B.frameOf (CB.as_mbuf c) == HyperStack.root}
```

Comment:
The reason gbuffer and glbuffer are annotated with the frame is that they are meant to apply only to "global" buffers. I am trying to limit the scope of these buffers and separate them from other heap- or stack-allocated buffers that also happen to be const and immutable. However, this particular annotation is not needed n the proofs so far. I wonder if knowing the location of the buffers will make certain other lemmas/optimizations possible in the futue.

Reply:
I think of "frameof == root" as the semantics of saying that something is a "global"

Reply:
I'm concerned that this will enrich the context with more region hypotheses about your buffers, which may in turn encourage z3 to apply a bunch of modifies-theory lemmas related to regions, and might be a waste of rlimit (and perhaps why you had to raise it in some places). So my advice would be that unless you know for sure that you need to think about regions, let's not clutter the context with them.